integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BaseDropReferenceExec.scala

```diff
@@ -21,24 +21,33 @@ import org.apache.spark.sql.connector.catalog.CatalogPlugin
 import org.apache.spark.sql.execution.datasources.v2.NessieUtils.unquoteRefName
 import org.apache.spark.unsafe.types.UTF8String
 import org.projectnessie.client.api.NessieApiV1
+import org.projectnessie.error.NessieReferenceNotFoundException
 
 abstract class BaseDropReferenceExec(
     output: Seq[Attribute],
     branch: String,
     currentCatalog: CatalogPlugin,
     isBranch: Boolean,
-    catalog: Option[String]
+    catalog: Option[String],
+    failOnDrop: Boolean
 ) extends NessieExec(catalog = catalog, currentCatalog = currentCatalog) {
 
   override protected def runInternal(
       api: NessieApiV1
```

Comment:
Just wondering: Can we switch to API v2 here (separate PR, of course)?

Reply:
Keeping in mind that `NessieAPIV2` extends `NessieAPIV1`, what is the immediate gain of forcing this code to only work with v2? From my (little) experience, Iceberg 1.3.1 is still on v1, right? Wouldn't your suggestion create issues with Iceberg < 1.4?

Reply:
This should work in Iceberg 1.3 too, AFAIK. API v2 has been around for a long time... but I did not check specifically :)

Reply:
I mean `NessieCatalog` in Iceberg 1.3 does use API v1, but v2 client should be available, I guess.

Reply:
switching to Nessie V2 client is not possible from Iceberg 1.3.

Possible from Iceberg 1.4 with the catalog property. 
ref: https://github.com/apache/iceberg/pull/6712

Reply:
let's defer this until Iceberg 1.4 is released :)